classes/output/core_renderer.php

```diff
@@ -463,4 +463,62 @@ public function block(\block_contents $bc, $region) {
 
         return $this->render_from_template('core/block', $context);
     }
+
+    /**
+     * Allow plugins to provide some content to be rendered in the navbar.
+     * The plugin must define a PLUGIN_render_navbar_output function that returns
+     * the HTML they wish to add to the navbar.
+     *
+     * @return string HTML for the navbar
+     */
+    public function navbar_plugin_output() {
+        $output = parent::navbar_plugin_output();
+        return $this->get_favourites() . $output;
+    }
+
+    /**
+     * Function that fetches all favorite courses, and renders them as a popover menu.
+     *
+     * @return string HTML to display the main header.
+     */
+    private function get_favourites() {
+        global $USER, $DB;
+        // Menu is relevant only for logged in users.
+        if (isloggedin()) {
+            $settings = get_config('theme_boost_union');
+            if (!isset($settings->shownavbarstarredcourses) || $settings->shownavbarstarredcourses == 'no') {
+                return '';
+            }
+            // Get all favourite courses.
+            $ufservice = \core_favourites\service_factory::get_service_for_user_context(\context_user::instance($USER->id));
+            $favourites = $ufservice->find_favourites_by_type('core_course', 'courses');
+            if (!$favourites) {
+                return '';
+            }
+            $favouritecourseids = array_map(
+                function($favourite) {
+                    return $favourite->itemid;
+                }, $favourites);
+            $courses = $DB->get_records_list('course', 'id', $favouritecourseids, 'visible DESC,sortorder ASC', 'id, shortname, fullname, visible');
```

Comment:
Span this line over two lines (max length 132 characters).

Reply:
And do resolve
```
514 | ERROR   | [x] Expected 1 space after "=>"; 0 found (Squiz.WhiteSpace.OperatorSpacing.NoSpaceAfter)
 514 | WARNING | [x] Short array syntax must be used to define arrays (Generic.Arrays.DisallowLongArraySyntax.Found)
 516 | ERROR   | [x] There should be a comma after the last array item in a multi-line array.
     |         |     (NormalizedArrays.Arrays.CommaAfterLast.MissingMultiLine)
```

Reply:
Hi @lucaboesch ,

Thanks for the feedback! The codechecker issues are fixed, and I included two behat tests that make sure the popover menu actually works. I based the code on MOODLE_403_STABLE branch, and squashed the commits into one (for now, waiting to see the results of the codechecker :)) 

Cheers,
Simeon